saleor/plugins/webhook/tasks.py

```diff
@@ -158,96 +185,172 @@ def send_webhook_using_google_cloud_pubsub(
+                response.content,
+            )
             try:
                 countdown = self.retry_backoff * (2 ** self.request.retries)
                 self.retry(countdown=countdown, **self.retry_kwargs)
             except MaxRetriesExceededError:
                 task_logger.warning(
                     "[Webhook] Failed request to %r: exceeded retry limit.",
-                    target_url,
+                    webhook.target_url,
                 )
+                delivery_status = EventDeliveryStatus.FAILED
+        delivery_update(delivery, delivery_status)
         task_logger.info(
             "[Webhook ID:%r] Payload sent to %r for event %r",
-            webhook_id,
-            target_url,
-            event_type,
+            webhook.id,
+            webhook.target_url,
+            delivery.event_type,
         )
-    else:
-        raise ValueError("Unknown webhook scheme: %r" % (parts.scheme,))
+    except ValueError as e:
+        response = WebhookResponse(content=str(e))
+        attempt_update(attempt, response)
+        delivery_update(delivery=delivery, status=EventDeliveryStatus.FAILED)
 
 
-def send_webhook_request_sync(app_name, target_url, secret, event_type, data: str):
-    parts = urlparse(target_url)
+def send_webhook_request_sync(app_name, delivery):
+    event_payload = delivery.payload
+    data = event_payload.payload
+    webhook = delivery.webhook
+    parts = urlparse(webhook.target_url)
     domain = Site.objects.get_current().domain
     message = data.encode("utf-8")
-    signature = signature_for_payload(message, secret)
+    signature = signature_for_payload(message, webhook.secret_key)
 
+    response = WebhookResponse(content="")
     response_data = None
     if parts.scheme.lower() in [WebhookSchemes.HTTP, WebhookSchemes.HTTPS]:
         logger.debug(
-            "[Webhook] Sending payload to %r for event %r.", target_url, event_type
+            "[Webhook] Sending payload to %r for event %r.",
+            webhook.target_url,
+            delivery.event_type,
         )
+        attempt = create_attempt(delivery=delivery, task_id=None)
         try:
             with webhooks_opentracing_trace(
-                event_type, domain, sync=True, app_name=app_name
+                delivery.event_type, domain, sync=True, app_name=app_name
             ):
                 response = send_webhook_using_http(
-                    target_url,
+                    webhook.target_url,
                     message,
                     domain,
                     signature,
-                    event_type,
+                    delivery.event_type,
                     timeout=WEBHOOK_SYNC_TIMEOUT,
                 )
-                response_data = response.json()
+                response_data = json.loads(response.content)
         except RequestException as e:
-            logger.warning("[Webhook] Failed request to %r: %r.", target_url, e)
+            logger.warning(
```

Comment:
IMHO, adding webhook ID would be usefull here

Reply:
added more info